recipe-client-addon/lib/RecipeRunner.jsm

```diff
@@ -23,22 +24,33 @@ this.EXPORTED_SYMBOLS = ["RecipeRunner"];
 
 const log = LogManager.getLogger("recipe-runner");
 const prefs = Services.prefs.getBranch("extensions.shield-recipe-client.");
+const TIMER_NAME = "recipe-client-addon-run";
 
 this.RecipeRunner = {
   init() {
     if (!this.checkPrefs()) {
       return;
     }
 
-    let delay;
     if (prefs.getBoolPref("dev_mode")) {
-      delay = 0;
-    } else {
-      // startup delay is in seconds
-      delay = prefs.getIntPref("startup_delay_seconds") * 1000;
+      // Run right now in dev mode
+      this.run();
     }
 
-    setTimeout(this.start.bind(this), delay);
+    // Run once every `runInterval` wall-clock seconds.
+    // This is managed by setting a "last ran" timestamp, and running if it is
+    // more than `runInterval` seconds ago.
+    const runInterval = prefs.getIntPref("run_interval_seconds");
+    const timerManager = Cc["@mozilla.org/updates/timer-manager;1"]
+                         .getService(Ci.nsIUpdateTimerManager);
```

Comment:
Much like how it was recommended we lazy-load modules when possible, we should probably do this for services too. It looks like [`XPCOMUtils.defineLazyServiceGetter`](https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/XPCOMUtils.jsm#defineLazyServiceGetter()) can do this for us. 

Also, can this be in the import section instead of within this function?